### PR TITLE
feat: Add tests

### PR DESCRIPTION
**Pull request by pborrego**
## What's New?

Adds the ability to write unit tests for the codebase. Adds basic high level tests around the intended public API of the MagnetAPIClient class. Documenting the expected interaction between the Magnet API and this binding class.

## How To Test?
1. Run locally
2. Run `npm test`

### CODE REVIEW

**Comment by sribatchu** [comment]
The tests are similar to https://github.com/cnnlabs/MagnetApiClient.Node/pull/1, however, adding my observations to this PR as well.

File: magnetapiclient.test.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    });
+
+    it('should be able to add an article', function (done) {
+
+        const request = {
+            text: 'SAMPLE ARTICLE TEXT',
+            title: 'SAMPLE ARTICLE TITLE',
+            insertDate: '23 JAN 2017 10:12:00 +01:00',
+            url: 'http://demo.klangoo.com/article-demo/api-example',
+            articleUID,
```
Comment (sribatchu) [inline magnetapiclient.test.js:35]: missing the value

File: magnetapiclient.test.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            done();
+        });
+    });
+
+    it('should be able to update an article', function (done) {
+
+        const request = {
+            text: 'SAMPLE ARTICLE TEXT UPDATED',
+            updateDate: '24 JAN 2017 10:12:00 +01:00',
+            articleUID,
```
Comment (sribatchu) [inline magnetapiclient.test.js:70]: seems to be missing value for this key

File: magnetapiclient.test.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        });
+    });
+
+    it('should be able to add an article', function (done) {
+
+        const request = {
+            text: 'SAMPLE ARTICLE TEXT',
+            title: 'SAMPLE ARTICLE TITLE',
+            insertDate: '23 JAN 2017 10:12:00 +01:00',
+            url: 'http://demo.klangoo.com/article-demo/api-example',
```
Comment (sribatchu) [inline magnetapiclient.test.js:34]: Only "text" or "url" is accepted in request by Magnet. Using both will cause Magnet API throw error.

File: magnetapiclient.test.js (hunk 2 above)
Comment (pborrego) [reply magnetapiclient.test.js:70]: It is just es6 shorthand for
```
{
    articleUID: articleUID
}
```

File: magnetapiclient.test.js (hunk 1 above)
Comment (pborrego) [reply magnetapiclient.test.js:35]: It is just es6 shorthand for
```
{
    articleUID: articleUID
}
```

File: magnetapiclient.test.js (hunk 3 above)
Comment (pborrego) [reply magnetapiclient.test.js:34]: Hmm that is good to know though their sample code that this test is based off of did the same thing: https://github.com/cnnlabs/MagnetApiClient.Node/blob/master/newsagencysample.js#L16

**Pull request closed without merging**